mbuild/formats/gomc_conf_writer.py

```diff
@@ -0,0 +1,4065 @@
+
+        # Check if charmm_object is really a Charmm() object
+        if isinstance(charmm_object, mf_charmm.Charmm) == False:
+            self.all_inputs_pass = False
+            print_error_message = 'The variable supplied as a charmm_object ({}}) is not a ' \
+                                  'charmm_object ({}})'.format(type(mf_charmm.Charmm), type(mf_charmm.Charmm))
+            raise ValueError(print_error_message)
+
+        # check ensemble is a correct type
+        print('INFO: ensemble_type = ' + str(ensemble_type))
+        if ensemble_type in ['NPT', 'NVT', 'GCMC', 'GEMC_NVT', 'GEMC_NPT']:
+            self.ensemble_type = ensemble_type
+        else:
+            self.all_inputs_pass = False
+            print_error_message = "The ensemble_type is not a valid entry. "\
+                                  "Only 'NPT', 'NVT', 'GCMC', 'GEMC_NVT', 'GEMC_NPT' are valid entries."
+            raise ValueError(print_error_message)
+
+        self.RunSteps = RunSteps
+        self.Temperature = Temperature
+        if charmm_object.FF_filename != None and isinstance(charmm_object.FF_filename, str) == True:
+            self.FF_filename = charmm_object.FF_filename
+        elif charmm_object.FF_filename is None or isinstance(charmm_object.FF_filename, str) == False:
+            self.all_inputs_pass = False
+            print_error_message = "The force field file name was not specified and in the Charmm object ({}})." \
+                                  "Therefore, the force field file (.inp) can not be written, and thus, the " \
+                                  "GOMC control file (.conf) can not be created. Please use the force field file " \
+                                  "name when building the Charmm object ({}})".format(type(mf_charmm.Charmm),
+                                                                                      type(mf_charmm.Charmm))
+            raise ValueError(print_error_message)
+
+        if charmm_object.filename_box_0 != None and isinstance(charmm_object.filename_box_0, str) == True:
+            self.Coordinates_box_0 = str(charmm_object.filename_box_0) + '.pdb'
+            self.Structures_box_0 = str(charmm_object.filename_box_0) + '.psf'
+        if charmm_object.filename_box_1 != None and isinstance(charmm_object.filename_box_1, str) == True:
+            self.Coordinates_box_1 = str(charmm_object.filename_box_1) + '.pdb'
+            self.Structures_box_1 = str(charmm_object.filename_box_1) + '.psf'
+        else:
+            self.Coordinates_box_1 = None
+            self.Structures_box_1 = None
+
+        self.coul_1_4_scaling = charmm_object.coul_1_4
+        self.input_variables_dict = input_variables_dict
+        self.residues_List = charmm_object.residues
+        self.all_atom_names_and_res_pairs_dict = charmm_object.all_atom_name_res_pairs_dict
+        self.all_atom_names_and_res_pairs_keys_List = list(self.all_atom_names_and_res_pairs_dict.keys())
```

Comment:
unless you are using the list of keys a lot, it does not seem necessary to make a list of the keys of the dict as well as storing the dict itself since you can generate that list when needed. Also, if the `charmm_object.all_atom_name_res_pairs_dict` gets updated at all, then the `self.all_atom_names_and_res_pairs_keys_List` will not get updated and will fall out of sync.

Reply:
OK.  removed an put only where it is needed an not part of the object.